analysis/include/MCAnaHistos.h

```diff
@@ -0,0 +1,31 @@
+#ifndef MCANAHISTOS_H
+#define MCANAHISTOS_H
+
+// ROOT
+#include "TClonesArray.h"
+
+// HPSTR
+#include "HistoManager.h"
+#include "MCParticle.h"
+#include "MCTrackerHit.h"
+#include "MCEcalHit.h"
+#include <string>
+#include <vector>
+
+class MCAnaHistos : public HistoManager {
+
+    public:
+
+        MCAnaHistos(const std::string& inputName) : HistoManager(inputName) { m_name = inputName; };
+
+        virtual void Define3DHistos(){};
+        virtual void Define2DHistos(){};
+        virtual void Define1DHistos();
+
+        void FillMCParticles(TClonesArray* mcParts, float weight = 1.);
+        void FillMCTrackerHits(TClonesArray* mcTrkrHits, float weight = 1.);
+        void FillMCEcalHits(TClonesArray* mcEcalHits, float weight = 1.);
```

Comment:
As personal taste we should use std::vectors instead of TClonesArray.

Reply:
Is the preference for usability or performance reasons, or just familiarity?

Reply:
Hi Maurik, 

I found some difficulties when I had to store the hits on track using TClonesArrays, while I found it straightforward when using std:vectors. In particular the current processor scheme lead to the following:
- Use a processor to dump all selected objects of type 1 to the event
- Use a second processor that uses (part of) type 1 objects to form type 2 objects and store them and create TRef between those. 

If instead you want to create type2 objects and only write the type1 objects related to them, then found issues with TClonesArray. 
This was the case I encountered when storing only the hits on track -> I didn't want *all* the hits to be stored.

Cheers,
PF